src/components/Tables/Table.tsx

```diff
@@ -132,28 +197,140 @@ const Table: React.FC<Props> = (props) => {
         }
     };
 
-    if (!domLoaded) {
-        return <></>;
-    }
+    const filters = (props.filters) ? Object.fromEntries(props.filters.map(filter => [filter, props.headers[filter]])) : props.headers;
```

Comment:
what does this do? potential to split out a little bit to be more clear - even if the actual 'code content' is still long, because JS is sometimes just bad!!